samcli/commands/deploy/auth_utils.py

```diff
@@ -63,7 +63,16 @@ def _auth_resource_event(sam_functions, sam_function, auth_resource_list):
                 if event.get("Properties", {}).get("Auth", False):
                     auth_resource_list.append((sam_function.name, True))
                 # Is there any auth defined on the referred http api or serverless api through the `id` construct?
-                elif _auth_id(sam_functions.resources, event.get("Properties", {}), identifier):
+                elif _auth_id(
+                    # use the resources containing the sam_function
+                    [
+                        resources
+                        for stack, resources in sam_function_provider.stack_and_resources
+                        if stack.stack_path_for_children_resources == sam_function.stack_path
+                    ][0],
```

Comment:
Again same usage, can we populate the array first and then check if it has more than zero element?

Reply:
Moved to `SamBuildStackProvider`